### PR TITLE
Update sendMessage modal style

### LINKED ISSUE #431: Message Modal: improve Send mode
Definition of done:
 - send mode should be a select ocupying 50% of the modal in terms of width
 - if Number of hops is chosen, then in the right side the number of hops textfield should show:
image
- if path is chosen, then a texfield similar to the one in 'import .csv channels to open' should show up under the select:
image
- the path textfield should automatically enter a new line if a 'space' is pressed (instead of entering space in the field)

Unfortunetely the modal will jump, but unless there is abetter solution without opening up new popups, we need to stick with it for now

### PR DESCRIPTION
**Pull request by ItsLuized**
Fixes #431 

### Overview

This pull request introduces visual improvementes to the sendMessage Modal

image
image
image
image
---
#### The select:
image

### CODE REVIEW

**Review by mjadach-iv — CHANGES_REQUESTED** [review]
- For the Number of hops -> 0 hop mesasge, enable the sand button and the the message using direct path
- For Path option, update the helpertext to have the new peerId types starting with 12, and also make it have new lines after commas to be more clear

**Review by mjadach-iv — CHANGES_REQUESTED** [review]
0 hop doesnt work:
{"tag":1,"body":"test2","peerId":"12D3KooWFSxmWxKLVRBnUb8G76Scd9ZTLYtBZjLmgPz429LJbArA","path":[],"hops":0}

direct works:
{"tag":1,"body":"test1","peerId":"12D3KooWFSxmWxKLVRBnUb8G76Scd9ZTLYtBZjLmgPz429LJbArA","path":[]}

File: src/components/Modal/node/SendMessageModal.tsx [hunk 1]
```diff
@@ -147,14 +159,29 @@ export const SendMessageModal = (props: SendMessageModalProps) => {
...
+      const input = e.target as HTMLInputElement;
+      // These are always a number... never null
+      const start = input.selectionStart!;
+      const end = input.selectionEnd!;
+      set_path((prevPath) => prevPath.substring(0, start) + '\n' + prevPath.substring(end));
+      // Set a timeout to ensure that the focus and blur events have a chance to complete
+      setTimeout(() => {
+        input.setSelectionRange(start + 1, start + 1);
+      }, 0);
+    }
```
Comment (mjadach-iv) [inline src/components/Modal/node/SendMessageModal.tsx:182]:
```suggestion
    else if (e.key === ',' ) {
      e.preventDefault(); // Prevent space key from inserting a space
      const input = e.target as HTMLInputElement;
      // These are always a number... never null
      const start = input.selectionStart!;
      const end = input.selectionEnd!;
      set_path((prevPath) => prevPath.substring(0, start) + ',\n' + prevPath.substring(end));
      // Set a timeout to ensure that the focus and blur events have a chance to complete
      setTimeout(() => {
        input.setSelectionRange(start + 1, start + 1);
      }, 0);
    }
```

Comment (mjadach-iv) [reply src/components/Modal/node/SendMessageModal.tsx:182]: also add new line for `,`
This should work, please test

**Review by mjadach-iv — CHANGES_REQUESTED** [review]

File: src/components/Modal/node/SendMessageModal.tsx (hunk 1 above)
Comment (ItsLuized) [reply src/components/Modal/node/SendMessageModal.tsx:181]: Selection range had to be adjusted, but good for everything else! 👍🏻

**Review by mjadach-iv — APPROVED** [review]

**Pull request merged**